### DISCUSSION TITLE
Are server redirects temporary or permanent?

### DISCUSSION THREAD

**Discussion by codinginflow** [vercel/next.js#50041] [App Router] [answered]
In the past, when we redirected in `getStaticProps` or `getServerSideProps`, we could set a `permanent` flag to indicate if this is a 301 or 302 redirect.

The new `redirect` function in the app router doesn't have this flag. Example:

```
export default async function Page({ searchParams }: PageProps) {
    const page = parseInt(searchParams.page?.toString() || "1");

    if (page < 1) {
        searchParams.page = "1";
        redirect("/blog?" + stringify(searchParams), RedirectType.replace);
    }
[...]
```

**Is this server-side redirect temporary or permanent?**

The documentation comment doesn't say anything about this. Just: 

> When used in a React server component, this will insert a meta tag to redirect the user to the target page.

**Comment by icyJoseph** [comment, marked as answer]
It is a 307 :) 

https://github.com/vercel/next.js/blob/canary/packages/next/src/server/app-render/app-render.tsx#L1508:

```ts
          if (isRedirectError(err)) {
            res.statusCode = 307
```

Though, we shouldn't have to look at the source code, nor setup a dummy project just to get this info. This should be documented, and modifiable I believe.

**Reply by icyJoseph** [reply]
Interestingly, I found this bit in another part of the source code:


```ts
/**
 * When used in a React server component, this will insert a meta tag to
 * redirect the user to the target page. When used in a custom app route, it
 * will serve a 302 to the caller.
 *
 * @param url the url to redirect to
 */
export function redirect(
  url: string,
  type: RedirectType = RedirectType.replace
): never {
  throw getRedirectError(url, type)
}
```

🤔

**Reply by codinginflow** [reply]
Thank you for your answer. Regarding your 2nd comment, what do you think that means?

**Reply by icyJoseph** [reply]
I think they are preparing some more features... since the docs still claim one param only, but it looks like you could redirect as push or replace, but no idea if that is trying to mimic browser history navigation methods, or if it is unrelated.

Hopefully, we get more features/docs in the upcoming days 😅

**Reply by codinginflow** [reply]
Thank you for your help, Joseph.
